ci/docker.sh

```diff
@@ -29,46 +32,95 @@ fi
 ####
 cd "${workdir}"
 
-if [ -z "${docker_hub_token:-}" ]; then
+if [[ -z "${docker_hub_token:-}" ]]; then
   fn_die "ERROR: DOCKER_HUB_TOKEN variable is not set. Exiting ..."
 fi
 
-if [ -z "${docker_hub_username:-}" ]; then
+if [[ -z "${docker_hub_username:-}" ]]; then
   fn_die "ERROR: DOCKER_HUB_USERNAME variable is not set. Exiting ..."
 fi
 
 docker_tag_full=""
-if [ "${is_a_release}" = "true" ]; then
-  docker_tag_full="${github_ref_name}"
-fi
+publish_tags=()
 
-# Building and publishing docker image
-if [ -n "${docker_tag_full:-}" ]; then
-  log_info "=== Building Docker image: ${docker_hub_org}/${docker_image_build_name}:${docker_tag_full} ==="
-  docker build --build-arg PROFILE=production -f "${docker_file_path}" -t "${docker_hub_org}/${docker_image_build_name}:${docker_tag_full}" .
-
-  # Publishing to DockerHub
-  log_info "=== Publishing Docker image(s) on Docker Hub ==="
-  echo "${docker_hub_token}" | docker login -u "${docker_hub_username}" --password-stdin
+if [[ "${is_a_release}" == "true" ]]; then
+  docker_tag_full="${github_ref_name}"
 
-  # Docker image(s) tags for PROD vs DEV release
-  if [ "${prod_release}" = "true" ]; then
+  # Determine image tags based on release type
+  if [[ "${prod_release}" == "true" ]]; then
     docker_tag_node="$(cut -d '-' -f1 <<< "${docker_tag_full}")"
     publish_tags=("${docker_tag_full}" "${docker_tag_node}" "latest")
-  elif [ "${dev_release}" = "true" ]; then
+  elif [[ "${dev_release}" == "true" ]]; then
     docker_tag_node="$(cut -d '-' -f1 <<< "${docker_tag_full}")-$(cut -d '-' -f3- <<< "${docker_tag_full}")"
     publish_tags=("${docker_tag_full}" "${docker_tag_node}")
-  elif [ "${test_release}" = "true" ]; then
+  elif [[ "${test_release}" == "true" ]]; then
     publish_tags=("${docker_tag_full}")
+  else
+    fn_die "ERROR: No valid release type specified for IS_A_RELEASE=true."
```

Comment:
We probably do not need this condition because setup_env.sh will never FINISH with IS_RELEASE=true, AND not matching prod/dev/test release. However, we can leave it here for extra-ordinary situation